snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/tracker/events/AbstractEvent.java

```diff
@@ -175,11 +182,30 @@ protected AbstractEvent() {
     }
 
     /**
-     * @return the events timestamp
+     * Get the timestamp of the event.
+     * @apiNote If the timestamp is not set, it sets one as a side effect.
+     * @deprecated As of release 1.5.0, it will be removed in the version 2.0.0.
+     * @return the event timestamp
      */
     @Override
+    @Deprecated
     public long getDeviceCreatedTimestamp() {
-        return this.deviceCreatedTimestamp;
+        if (deviceCreatedTimestamp == null) {
```

Comment:
I'm not sure why we have introduced a side effect in this function and then made a new function to get the unchanged version. Whats the plan for v2?